x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/page.tsx

```diff
@@ -16,29 +18,59 @@ import { DataFrameAnalysisConfigType } from '../../../../../common/types/data_fr
 import { HelpMenu } from '../../../components/help_menu';
 import { useMlKibana } from '../../../contexts/kibana';
 import { MlPageHeader } from '../../../components/page_header';
+import { AnalyticsIdSelector } from '../components/analytics_selector';
 
 export const Page: FC<{
   jobId: string;
   analysisType: DataFrameAnalysisConfigType;
 }> = ({ jobId, analysisType }) => {
+  const [analyticsId, setAnalyticsId] =
+    useState<{ model_id?: string; job_id?: string; analysis_type?: string }>();
   const {
     services: { docLinks },
   } = useMlKibana();
   const helpLink = docLinks.links.ml.dataFrameAnalytics;
+  const jobIdToUse = jobId || analyticsId?.job_id;
+  const analysisTypeToUse = analysisType || analyticsId?.analysis_type;
+
   return (
     <>
-      <div data-test-subj="mlPageDataFrameAnalyticsExploration">
-        <MlPageHeader>{jobId}</MlPageHeader>
-        {analysisType === ANALYSIS_CONFIG_TYPE.OUTLIER_DETECTION && (
-          <OutlierExploration jobId={jobId} />
-        )}
-        {analysisType === ANALYSIS_CONFIG_TYPE.REGRESSION && (
-          <RegressionExploration jobId={jobId} />
-        )}
-        {analysisType === ANALYSIS_CONFIG_TYPE.CLASSIFICATION && (
-          <ClassificationExploration jobId={jobId} />
-        )}
-      </div>
+      <MlPageHeader>
+        <FormattedMessage
+          id="xpack.ml.dataframe.analyticsExploration.title"
+          defaultMessage="Explore results for Analytics ID {id}"
```

Comment:
If I open the results page from the side nav, the `id` doesn't seem to be set in the page header:

![image](https://user-images.githubusercontent.com/7405507/157268704-147a5aa2-eca3-49d3-b7be-c49c2819a460.png)

Reply:
Updated in d14681d364db8c271d99c348c164e10426f53982